include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp

```diff
@@ -131,21 +138,21 @@ alignment_trace(database_t && database,
         if ((dir & L) == L)
         {
             col = std::max<size_t>(col, 1) - 1;
-            gapped_database.push_front(database[col]);
-            gapped_query.push_front(gap{});
+            --it_db;
```

Comment:
It is not guaranteed that you can decrement the end iterator! It could be a sentinel and no iterator.

Reply:
Also see https://en.cppreference.com/w/cpp/iterator/prev
(Would it work on a sentinel though?)

Reply:
This is ok then.